src/System.Windows.Extensions/src/System/Security/Cryptography/X509Certificates/SafeX509Handles.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal sealed class SafeCertContextHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        // 0 is an Invalid Handle
+        internal SafeCertContextHandle(IntPtr handle) : base(true)
+        {
+            SetHandle(handle);
+        }
+
+        internal static SafeCertContextHandle InvalidHandle => new SafeCertContextHandle(IntPtr.Zero);
```

Comment:
This is allocating a new handle every time the property is accessed.  I've not looked at the call sites to see if that's really needed, but if it's not, we should consider caching it, and if it is, we should make this a `CreateInvalidHandle()` method or something like that, so that it's clear that accessing the property allocates a new object.

Reply:
See also: Common\Microsoft\Win32\SafeHandles\SafeHandleCache.cs